src/main/java/databaseInteracting/DataBaseInteracting.java

```diff
@@ -0,0 +1,173 @@
+    String sql = "SELECT * FROM products";
+    return connection.prepareStatement(sql);
+  }
+
+  /**
+   * Method to add a product on the products table.
+   *
+   * @param name  String - name of the product.
+   * @param stock int - number of products on the stock.
+   * @param price float - price of the product.
+   * @throws SQLException - Exception of database.
+   */
+  public void addProductToDatabase(String name, int stock, float price) throws SQLException {
+    String sql = "INSERT INTO products (name, stock, price) VALUES (?,?,?)";
+    PreparedStatement preparedStatement = connection.prepareStatement(sql);
+    preparedStatement.setString(1, name);
+    preparedStatement.setInt(2, stock);
+    preparedStatement.setFloat(3, price);
+    preparedStatement.executeUpdate();
+    preparedStatement.close();
+  }
+
+  /**
+   * Method to get products by the id.
+   *
+   * @param id int - id of the product.
+   * @return a resultSet with the products.
+   * @throws SQLException - Exception of database.
+   */
+  public ResultSet getProductsById(int id) throws SQLException {
+    PreparedStatement preparedStatement =
+        connection.prepareStatement("select * from products where id = ?");
+    preparedStatement.setInt(1, id);
+    return preparedStatement.executeQuery();
+  }
+
+  /**
+   * Method to get a resultSet of products table passing a Prepared Statement.
+   *
+   * @param preparedStatement PreparedStatement -
+   *                          prepared statement of the query to list all products.
+   * @return resultSet with data of the products table.
+   * @throws SQLException - Exception of database.
+   */
+  public ResultSet createResultSet(PreparedStatement preparedStatement) throws SQLException {
+    return preparedStatement.executeQuery();
+  }
+
+  /**
+   * Method to return all products into a Products table.
+   *
+   * @return list of products.
+   * @throws SQLException Exception of database.
+   */
+  public List<databaseInteracting.Product> searchForAllProductsOnDatabase() throws SQLException {
+    List<databaseInteracting.Product> list = new ArrayList<>();
+    try (PreparedStatement preparedStatement = createPrepareStatementToProductList();
+         ResultSet aResultSet = createResultSet(preparedStatement)) {
+      PrepareAndExecuteMethod(aResultSet, list);
+    }
+    return list;
+  }
+  /*
+  /**
+   * Method to make a query into the shopping cart table.
+   * @return a list of products from shopping cart table.
+   * @throws SQLException - Exception of database.
+  */
+  public ResultSet searchForAllProductsOnShoppingCart(int cartId) throws SQLException {
+    String sql = "SELECT p.id, p.name, p.price, l.quantity FROM products p," +
+        " shoppinglist l, shoppingcart c where p.id = l.idproduct AND l.idcart = ?;";
+      PreparedStatement preparedStatement = connection.prepareStatement(sql);
+      preparedStatement.setInt(1,cartId);
+      return preparedStatement.executeQuery();
```

Comment:
Prepared statement isn't being closed here

Reply:
Used a try with resources in the class that invoke this method to close the prepared statement.